### PR TITLE
[5004] Invalid timezone causes Studio UI to crash

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/5004

### CODE REVIEW

File: ui/app/src/state/epics/configuration.ts [hunk 1]
```diff
@@ -46,13 +56,28 @@ export default [
...
+              const hostToHost$ = getHostToHostBus();
+              hostToHost$.next(
+                showSystemNotification({
+                  // @ts-ignore
+                  message: getIntl().formatMessage(configurationMessages.localeError, { message: e.message })
+                })
+              );
+            }
+          }),
           map((config) => fetchSiteConfigComplete(config)),
```
Comment (rart) [inline ui/app/src/state/epics/configuration.ts]: Why ts-ignore? Should need to be one. **ALWAYS** explain any ts/eslint ignores you add.
```suggestion
          switchMap((config) => {
            try {
              let localeCode = config.locale?.localeCode || state.uiConfig.locale.localeCode;
              let options = config.locale?.dateTimeFormatOptions || state.uiConfig.locale.dateTimeFormatOptions;
              new Intl.DateTimeFormat(localeCode, options);
              return [fetchSiteConfigComplete(config)];
            } catch (e) {
                return [
                  fetchSiteConfigComplete({ ...config, /* use modified fallback values here and remove try from reducer */ }), 
                  showSystemNotification({
                    message: getIntl().formatMessage(configurationMessages.localeError, { message: e.message })
                  })
                ];
              );
            }
          }),
```

File: ui/app/src/state/reducers/uiConfig.ts [hunk 2]
```diff
@@ -148,14 +148,25 @@ const reducer = createReducer<GlobalState['uiConfig']>(initialState, {
...
+    let localeCode = locale?.localeCode || initialState.locale.localeCode;
+    let options = locale?.dateTimeFormatOptions || initialState.locale.dateTimeFormatOptions;
+
+    // check for locale and options validity
+    try {
+      new Intl.DateTimeFormat(localeCode, options);
+    } catch (e) {
+      localeCode = initialState.locale.localeCode;
+      options = initialState.locale.dateTimeFormatOptions;
+    }
```
Comment (rart) [inline ui/app/src/state/reducers/uiConfig.ts]: I think we can now roll this back as the epic cleanses

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/state/epics/configuration.ts [hunk 3]
```diff
@@ -46,14 +55,37 @@ export default [
...
+                  locale: {
+                    ...config.locale,
+                    localeCode: state.uiConfig.locale.localeCode,
+                    dateTimeFormatOptions: state.uiConfig.locale.dateTimeFormatOptions
+                  }
+                }),
+                showSystemNotification({
+                  // @ts-ignore - 'message' does not exist on type 'unknown', and catch variable type annotation must
+                  // be 'any' or 'unknown', but setting it to 'any' causes a prettier error.
+                  message: getIntl().formatMessage(configurationMessages.localeError, { message: e.message })
```
Comment (rart) [inline ui/app/src/state/epics/configuration.ts]:
```suggestion
                  message: getIntl().formatMessage(configurationMessages.localeError, { message: (e as Error).message })
```
Oddly, it works even without the casting for me. I didn't see the `'message' does not exist on type 'unknown'` TS error via the IDE or the compilation even removing the TS ignore. Do you get the error when compiling?

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/state/epics/configuration.ts (hunk 3 above)
Comment (jvega190) [reply ui/app/src/state/epics/configuration.ts]: I just realized that the error is being displayed only on the IDE, not when compiling. Should I leave it with the casting?

**Review by rart — APPROVED** [review]

**Pull request merged**